pkg/httpchecker/reattempt.go

```diff
@@ -0,0 +1,35 @@
+package httpchecker
+
+import (
+	"math/rand"
+	"time"
+)
+
+func init() {
+	rand.New(rand.NewSource(time.Now().UnixNano())) //nolint:gosec
+}
+
+type stop struct {
+	error
+}
```

Comment:
Does this need to a struct separately if we only have error here?

Reply:
Yes this is not needed but I have keep it consistent as same and ref from MUO repo